File: app/src/main/java/it/paranoidsquirrels/idleguildmaster/storage/data/places/raids/TheDireDescent.java
package it.paranoidsquirrels.idleguildmaster.storage.data.places.raids;

import it.paranoidsquirrels.idleguildmaster.AchievementsUtils;
import it.paranoidsquirrels.idleguildmaster.MainActivity;
import it.paranoidsquirrels.idleguildmaster.R;
import it.paranoidsquirrels.idleguildmaster.Utils;
import it.paranoidsquirrels.idleguildmaster.databinding.LayoutDungeonBinding;
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy;
import it.paranoidsquirrels.idleguildmaster.storage.data.places.Area;
import it.paranoidsquirrels.idleguildmaster.storage.data.places.Logger;
import java.util.Arrays;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.concurrent.CopyOnWriteArrayList;

/* JADX INFO: loaded from: classes3.dex */
public class TheDireDescent extends Area {
    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.places.Area
    public int adventurersNumber() {
        return 8;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.places.Area
    public int getAreaType() {
        return 2;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.places.Area
    public int getDarkness() {
        return 0;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.places.Area
    public int getName() {
        return R.string.raid_name_the_dire_descent;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.places.Area
    public int getSummaryDrawable() {
        return R.drawable.summary_the_dire_descent;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.places.Area
    public int getDetailDrawable() {
        return R.drawable.area_the_dire_descent;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.places.Area
    public LayoutDungeonBinding getLayout() {
        return MainActivity.raidsFragment.getBinding().theDireDescent;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.places.Area
    protected List<Enemy> rollEnemies() {
        if (this.progress < this.maxProgress) {
            return new CopyOnWriteArrayList();
        }
        if (this.progress != 5) {
            return new CopyOnWriteArrayList();
        }
        return Utils.gotUniqueDrop("SerpentLunge", this) ? new CopyOnWriteArrayList() : new CopyOnWriteArrayList(Arrays.asList(Enemy.getInstance("HeraldXavi"), Enemy.getInstance("HeraldMaya"), Enemy.getInstance("HeraldShoran")));
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.places.Area
    protected void searchRoom() {
        Logger.log(this, 41, new Object[0]);
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.places.Area
    protected void triggerEvent(String str) {
        str.hashCode();
        switch (str) {
            case "enter_room":
                switch (this.progress) {
                    case 1:
                        Logger.log(this, 100, Integer.valueOf(R.string.log_the_dire_descent_room_1));
                        break;
                    case 2:
                        Logger.log(this, 100, Integer.valueOf(R.string.log_the_dire_descent_room_2));
                        break;
                    case 3:
                        Logger.log(this, 100, Integer.valueOf(R.string.log_the_dire_descent_room_3));
                        break;
                    case 4:
                        Logger.log(this, 100, Integer.valueOf(R.string.log_the_dire_descent_room_4));
                        break;
                    case 5:
                        Logger.log(this, 100, Integer.valueOf(R.string.log_the_dire_descent_room_5));
                        break;
                    case 6:
                        Logger.log(this, 100, Integer.valueOf(R.string.log_the_dire_descent_room_6));
                        AchievementsUtils.unlock(AchievementsUtils.ACHIEVEMENT_THE_CORE);
                        break;
                    case 7:
                        Logger.log(this, 100, Integer.valueOf(R.string.log_the_dire_descent_room_7));
                        break;
                    case 8:
                        Logger.log(this, 100, Integer.valueOf(R.string.log_the_dire_descent_room_8));
                        this.terminationRequested = true;
                        break;
                }
                            case "fight_start":
                if (this.progress == 5) {
                    Logger.log(this, 101, Integer.valueOf(R.string.log_the_dire_descent_encounter_1));
                    break;
                }
                            case "enter_dungeon":
                Logger.log(this, 100, Integer.valueOf(R.string.log_the_dire_descent_enter));
                break;
        }
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.places.Area
    public LinkedHashMap<Area, Integer> listAreasUnlocked() {
        LinkedHashMap<Area, Integer> linkedHashMap = new LinkedHashMap<>();
        linkedHashMap.put(MainActivity.data.getSleepingPlanet(), 7);
        linkedHashMap.put(MainActivity.data.getKaunis(), 7);
        linkedHashMap.put(MainActivity.data.getTheTower(), 7);
        return linkedHashMap;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.places.Area
    public List<Enemy> listEnemies() {
        return Arrays.asList(Enemy.getInstance("HeraldXavi"), Enemy.getInstance("HeraldMaya"), Enemy.getInstance("HeraldShoran"));
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.places.Area
    public boolean completed() {
        return this.maxProgress >= 7 && this.drops.isEmpty();
    }
}

File: app/src/main/kotlin/it/paranoidsquirrels/idleguildmaster/storage/data/places/raids/TheDireDescent.kt
package it.paranoidsquirrels.idleguildmaster.storage.data.places.raids

import it.paranoidsquirrels.idleguildmaster.AchievementsUtils
import it.paranoidsquirrels.idleguildmaster.MainActivity
import it.paranoidsquirrels.idleguildmaster.R
import it.paranoidsquirrels.idleguildmaster.Utils
import it.paranoidsquirrels.idleguildmaster.databinding.LayoutDungeonBinding
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
import it.paranoidsquirrels.idleguildmaster.storage.data.places.Area
import it.paranoidsquirrels.idleguildmaster.storage.data.places.Logger
import java.util.LinkedHashMap
import java.util.concurrent.CopyOnWriteArrayList

class TheDireDescent : Area() {
    override fun adventurersNumber(): Int = 8

    override fun getAreaType(): Int = 2

    override fun getDarkness(): Int = 0

    override fun getName(): Int = R.string.raid_name_the_dire_descent

    override fun getSummaryDrawable(): Int = R.drawable.summary_the_dire_descent

    override fun getDetailDrawable(): Int = R.drawable.area_the_dire_descent

    override fun getLayout(): LayoutDungeonBinding = MainActivity.raidsFragment.binding.theDireDescent

    override fun rollEnemies(): MutableList<Enemy> {
        if (progress < maxProgress) {
            return CopyOnWriteArrayList()
        }
        if (progress != 5) {
            return CopyOnWriteArrayList()
        }
        return if (Utils.gotUniqueDrop("SerpentLunge", this)) CopyOnWriteArrayList() else CopyOnWriteArrayList(listOfNotNull(
            Enemy.getInstance("HeraldXavi"),
            Enemy.getInstance("HeraldMaya"),
            Enemy.getInstance("HeraldShoran")
        ))
    }

    override fun searchRoom() {
        Logger.log(this, 41)
    }

    override fun triggerEvent(str: String) {
        when (str) {
            "enter_room" -> {
                when (progress) {
                    1 -> Logger.log(this, 100, R.string.log_the_dire_descent_room_1)
                    2 -> Logger.log(this, 100, R.string.log_the_dire_descent_room_2)
                    3 -> Logger.log(this, 100, R.string.log_the_dire_descent_room_3)
                    4 -> Logger.log(this, 100, R.string.log_the_dire_descent_room_4)
                    5 -> Logger.log(this, 100, R.string.log_the_dire_descent_room_5)
                    6 -> {
                        Logger.log(this, 100, R.string.log_the_dire_descent_room_6)
                        AchievementsUtils.unlock(AchievementsUtils.ACHIEVEMENT_THE_CORE)
                    }
                    7 -> Logger.log(this, 100, R.string.log_the_dire_descent_room_7)
                    8 -> {
                        Logger.log(this, 100, R.string.log_the_dire_descent_room_8)
                        terminationRequested = true
                    }
                }
            }
            "fight_start" -> {
                if (progress == 5) {
                    Logger.log(this, 101, R.string.log_the_dire_descent_encounter_1)
                }
            }
            "enter_dungeon" -> {
                Logger.log(this, 100, R.string.log_the_dire_descent_enter)
            }
        }
    }

    override fun listAreasUnlocked(): LinkedHashMap<Area, Int> {
        val linkedHashMap = LinkedHashMap<Area, Int>()
        MainActivity.data.sleepingPlanet?.let { linkedHashMap[it] = 7 }
        MainActivity.data.kaunis?.let { linkedHashMap[it] = 7 }
        MainActivity.data.theTower?.let { linkedHashMap[it] = 7 }
        return linkedHashMap
    }

    override fun listEnemies(): List<Enemy> {
        return listOfNotNull(
            Enemy.getInstance("HeraldXavi"),
            Enemy.getInstance("HeraldMaya"),
            Enemy.getInstance("HeraldShoran")
        )
    }

    override fun completed(): Boolean {
        return maxProgress >= 7 && drops.isEmpty()
    }
}
